File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/PenetrationRune.java
package org.rexcellentgames.burningknight.entity.item.accessory.equippable;

import org.rexcellentgames.burningknight.assets.Locale;

public class PenetrationRune extends Equippable {
	{
		name = Locale.get("penetration_rune");
		description = Locale.get("penetration_rune_desc");
		sprite = "item-penetration_stone";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.penetrates = true;
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.penetrates = false;
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/PenetrationRune.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class PenetrationRune : Equippable() {
	init {
		name = Locale.get("penetration_rune")
		description = Locale.get("penetration_rune_desc")
		sprite = "item-penetration_stone"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.penetrates = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.penetrates = false
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}